backend/src/tasks/cve.ts

```diff
@@ -40,6 +49,9 @@ const identifyPassiveCVEs = async () => {
               cpes.add(cpe + ':' + product.version);
             }
           }
+          if (!servicesToCheck.has(product.cpe + ':' + product.version)) {
+            servicesToCheck.set(product.cpe + ':' + product.version, service);
```

Comment:
I don't think this will necessarily work, because a CPE / version isn't necessarily unique to a service (we might have multiple instances of the same CPE found, each which belongs to separate services).